### DISCUSSION TITLE
How is Configuration.php generated?

### DISCUSSION THREAD

**Discussion by smargoli2** [amazon-php/sp-api-sdk#247] [Q&A]
Hi,
We are generating C# code from the spi api models using the swagger open api generator. When we generate all the models, we get an API client for each model. I noticed that you have just one Configuration.php class for all the SDKs. How did you develop that code? Is it based on a generated class...? Can you advise as to how we should generate one client to use with all the SDKs (models)?

**Comment by smargoli2** [comment]
We figured out how to generate one ApiClient for our project using swagger, with the -Dmodels, -Dapis and/or -DsupportingFiles as parameters for the swagger command string.

**Comment by norberttech** [comment]
Hey, sorry I missed your question. 
It's actually pretty straightforward `Configuration.php` is outside of the Api/Models namespace (it's not removed when code is regenerated) and the reference to this class is used across all templates.
